src/pages/workspace/expensifyCard/WorkspaceEditCardLimitPage.tsx

```diff
@@ -0,0 +1,153 @@
+
+// TODO: remove when Onyx data is available
+const mockedCard = {
+    accountID: 885646,
+    availableSpend: 1000,
+    nameValuePairs: {
+        cardTitle: 'Test 1',
+        isVirtual: true,
+        limit: 2000,
+        limitType: CONST.EXPENSIFY_CARD.LIMIT_TYPES.SMART,
+    },
+    lastFourPAN: '1234',
+};
+
+type WorkspaceEditCardLimitPageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.EXPENSIFY_CARD_LIMIT>;
+
+function WorkspaceEditCardLimitPage({route}: WorkspaceEditCardLimitPageProps) {
+    const {policyID, cardID} = route.params;
+    const {translate} = useLocalize();
+    const {inputCallbackRef} = useAutoFocusInput();
+    const styles = useThemeStyles();
+    const [isConfirmModalVisible, setIsConfirmModalVisible] = useState(false);
+
+    const [cardsList] = useOnyx(`${ONYXKEYS.COLLECTION.WORKSPACE_CARDS_LIST}${policyID}_${CONST.EXPENSIFY_CARD.BANK}`);
+    const card = cardsList?.[cardID] ?? mockedCard;
+
+    const getPromptTextKey = useMemo((): ConfirmationWarningTranslationPaths => {
+        switch (card.nameValuePairs?.limitType) {
+            case CONST.EXPENSIFY_CARD.LIMIT_TYPES.SMART:
+                return 'workspace.expensifyCard.smartLimitWarning';
+            case CONST.EXPENSIFY_CARD.LIMIT_TYPES.FIXED:
+                return 'workspace.expensifyCard.fixedLimitWarning';
+            case CONST.EXPENSIFY_CARD.LIMIT_TYPES.MONTHLY:
+                return 'workspace.expensifyCard.monthlyLimitWarning';
+            default:
+                return 'workspace.expensifyCard.fixedLimitWarning';
+        }
+    }, [card.nameValuePairs?.limitType]);
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const updateCardLimit = (newLimit: string) => {
+        setIsConfirmModalVisible(false);
+
+        // TODO: add API call when it's supported https://github.com/Expensify/Expensify/issues/407831
+
+        Navigation.goBack();
+    };
+
+    const submit = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.EDIT_EXPENSIFY_CARD_LIMIT_FORM>) => {
+        const currentLimit = card.nameValuePairs?.limit ?? 0;
+        const currentSpend = currentLimit - card.availableSpend;
+        const newLimit = Number(values[INPUT_IDS.LIMIT]) * 100;
+        const newAvailableSpend = newLimit - currentSpend;
+
+        if (newAvailableSpend <= 0) {
+            setIsConfirmModalVisible(true);
+            return;
+        }
+
+        updateCardLimit(values[INPUT_IDS.LIMIT]);
+    };
+
+    const validate = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.EDIT_EXPENSIFY_CARD_LIMIT_FORM>): FormInputErrors<typeof ONYXKEYS.FORMS.EDIT_EXPENSIFY_CARD_LIMIT_FORM> => {
+            const errors = ValidationUtils.getFieldRequiredErrors(values, [INPUT_IDS.LIMIT]);
+
+            // We only want integers to be sent as the limit
+            if (!Number(values.limit) || !Number.isInteger(Number(values.limit))) {
```

Comment:
NAB: I think we shouldn't allow users type decimal (can't type "." character). It would be better

Reply:
The same approach is used on the LimitStep page, see the [explanation](https://github.com/Expensify/App/pull/44741#issuecomment-2213768449). So let's keep it this way for now